doc/languages-frameworks/lisp.section.md

```diff
@@ -0,0 +1,225 @@
+expression with package definitions in a file called `imported.nix`.
+
+To run this import process, execute:
+
+```sh
+$ nix-shell
+$ sbcl --script ql-import.lisp
+```
+
+It also creates a `packages.sqlite` file. It's used during the
+generation of the `imported.nix` file and can be safely removed. It
+contains the full information of Quicklisp packages, so you can use it
+to query the dependency graphs using SQL, if you're interested.
+
+### Choosing a Quicklisp release {#lisp-choosing-a-quicklisp-release}
+
+Quicklisp release URL is configured in the `import/main.lisp` file.
+
+Change the `:dist-url` keyword argument to `make-instance 'quicklisp-repository`
+before running `ql-import.lisp` to upgrade to a newer release.
+
+## Manually defining packages {#lisp-manually-defining-packages}
+
+Packages are declared using `buildASDFSystem`. This function takes
+the following arguments and returns a Lisp package derivation.
+
+### Example {#lisp-build-asdf-system-example}
+
+```nix
+sbcl.buildASDFSystem {
+  pname = "alexandria";
+  version = "v1.4";
+  src = pkgs.fetchFromGitLab {
+    domain = "gitlab.common-lisp.net";
+    owner = "alexandria";
+    repo = "alexandria";
+    rev = "v1.4";
+    hash = "sha256-1Hzxt65dZvgOFIljjjlSGgKYkj+YBLwJCACi5DZsKmQ=";
+  };
+}
+```
+
+
+### Required arguments {#lisp-build-asdf-system-required-arguments}
+
+- `pname`
+Name of the package/library
+
+- `version`
+Version of the package/library
+
+- `src`
+Source of the package/library (`fetchTarball`, `fetchGit`, `fetchMercurial` etc.)
+
+### Optional arguments {#lisp-build-asdf-system-optional-arguments}
+
+- `patches ? []`
+
+Patches to apply to the source code before compiling it. This is a
+list of files.
+
+- `nativeLibs ? []`
+
+Native libraries, will be appended to the library
+path. (`pkgs.openssl` etc.)
+
+- `javaLibs ? []`
+
+Java libraries for ABCL, will be appended to the class path.
+
+- `lispLibs ? []`
+
+Lisp dependencies These must themselves be packages built with
+`buildASDFSystem`
+
+- `systems ? [ pname ]`
+
+Some libraries have multiple systems under one project, for example,
+`cffi` has `cffi-grovel`, `cffi-toolchain` etc.  By default, only the
+`pname` system is build.
+
+`.asd's` not listed in `systems` are removed before saving the library
+to the Nix store. This prevents ASDF from referring to uncompiled
+systems on run time.
+
+Also useful when the `pname` is differrent than the system name, such
+as when using reverse domain naming. (see `jzon` -> `com.inuoe.jzon`)
+
+### Return value {#lisp-build-asdf-system-return-value}
+
+A `derivation` that, when built, contains the sources and pre-compiled
+FASL files (Lisp implementation dependent) alongside any other
+artifacts generated during compilation.
+
+## Building a Lisp with packages: `withPackages` {#lisp-building-a-lisp-with-packages}
+
+### Example {#lisp-building-a-lisp-with-packages-example}
+
+`sbcl.withPackages (ps: [ ps.alexandria ])`
```

Comment:
library examples should contain what is returned or parts of it

Reply:
Well, the resulting derivation is described afterwards, I think it is fine.